### PR TITLE
Always show context menu on right click

### LINKED ISSUE #2235: Right-click opens a context menu every other time when clicking on same row
https://user-images.githubusercontent.com/98249521/186052763-72f4ea17-88ac-46e5-ba11-441f60d7b088.mov

### PR DESCRIPTION
**Pull request by mattseddon**
Closes #2235

### Demo

https://user-images.githubusercontent.com/37993418/189105077-3db6d7ee-fbdd-4ff6-8230-cf5f4546d4f2.mov

https://user-images.githubusercontent.com/37993418/189105875-0c91bbaf-a9c1-4051-bc5a-db2f0d889985.mov

(this was painful)

### CODE REVIEW

File: webview/src/shared/components/contextMenu/ContextMenu.tsx [hunk 1]
```diff
@@ -58,6 +60,8 @@ export const ContextMenu: React.FC<ContextMenuProps> = ({
     placement={'bottom'}
     interactive
     onTrigger={positionContextMenuAndDisableEvents}
+    onClickOutside={(instance: Instance) => instance.hide()}
+    hideOnClick={false}
```
Comment (mattseddon) [inline webview/src/shared/components/contextMenu/ContextMenu.tsx:64]: [F] Get tippy to not close the tooltip for you if you click inside the row but instead move the context menu if you right click.

File: webview/src/shared/components/contextMenu/ContextMenu.tsx [hunk 2]
```diff
@@ -58,6 +60,8 @@ export const ContextMenu: React.FC<ContextMenuProps> = ({
     placement={'bottom'}
     interactive
     onTrigger={positionContextMenuAndDisableEvents}
+    onClickOutside={(instance: Instance) => instance.hide()}
```
Comment (mattseddon) [inline webview/src/shared/components/contextMenu/ContextMenu.tsx:63]: [F] If you click outside of the row hide the instance and make a new one

File: webview/src/shared/components/contextMenu/ContextMenu.tsx [hunk 3]
```diff
@@ -29,11 +29,13 @@ const positionContextMenuAndDisableEvents = (
...
 
   const hideOnClick = () => {
-    instance.hide()
+    !instance.state.isDestroyed && instance.hide()
   }
 
   instance.popper.removeEventListener('click', hideOnClick)
   instance.popper.addEventListener('click', hideOnClick)
+
+  window.addEventListener('click', hideOnClick, { once: true })
```
Comment (mattseddon) [inline webview/src/shared/components/contextMenu/ContextMenu.tsx:38]: [F] Left click anywhere should close the tooltip

**Review by sroy3 — APPROVED** [review]

**Comment by sroy3** [comment]
Maybe adding a few tests would help.

**Review by julieg18 — APPROVED** [review]
Great job!

**Review by wolmir — APPROVED** [review]

**Comment by mattseddon** [comment]
> Maybe adding a few tests would help.

:sad_face: ... ya got me [insert meme]

I'll write some today.

File: webview/src/shared/components/contextMenu/ContextMenu.tsx (hunk 2 above)
Comment (mattseddon) [reply webview/src/shared/components/contextMenu/ContextMenu.tsx:63]: A bug in Tippy stops this from working with `@testing-library/react`. I have spent ~1.5 hrs and cannot find a way around it. Two clicks in the storybook show the bug as well. IRL it works 😕.

**Pull request merged**